cfn-resources/alert-configuration/cmd/resource/resource.go

```diff
@@ -259,52 +270,52 @@ func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
 }
 
 // function to check if record already exist
-func isExist(currentModel *Model, client *mongodbatlas.Client) bool {
-	alert, _, err := client.AlertConfigurations.GetAnAlertConfig(context.Background(), *currentModel.ProjectId, *currentModel.Id)
+func isExist(currentModel *Model, client *atlasSDK.APIClient) bool {
+	alert, _, err := client.AlertConfigurationsApi.GetAlertConfiguration(context.Background(), *currentModel.ProjectId, *currentModel.Id).Execute()
 	return err == nil && alert != nil
 }
 
-func expandAlertConfigurationMatchers(matchers []Matcher) []mongodbatlas.Matcher {
-	mts := make([]mongodbatlas.Matcher, 0)
+func expandAlertConfigurationMatchers(matchers []Matcher) []map[string]interface{} {
+	mts := make([]map[string]interface{}, 0)
 	for ind := range matchers {
-		mMatcher := mongodbatlas.Matcher{
-			FieldName: cast.ToString(matchers[ind].FieldName),
-			Operator:  cast.ToString(matchers[ind].Operator),
-			Value:     cast.ToString(matchers[ind].Value),
+		mMatcher := map[string]interface{}{
+			"FieldName": cast.ToString(matchers[ind].FieldName),
+			"Operator":  cast.ToString(matchers[ind].Operator),
+			"Value":     cast.ToString(matchers[ind].Value),
 		}
 		mts = append(mts, mMatcher)
 	}
 	return mts
 }
 
-func expandAlertConfigurationMetricThresholdConfig(currentModel *Model) *mongodbatlas.MetricThreshold {
+func expandAlertConfigurationMetricThresholdConfig(currentModel *Model) *atlasSDK.ServerlessMetricThreshold {
 	threshold := currentModel.MetricThreshold
 	if threshold == nil {
 		return nil
 	}
-	return &mongodbatlas.MetricThreshold{
+	return &atlasSDK.ServerlessMetricThreshold{
 		MetricName: cast.ToString(threshold.MetricName),
-		Operator:   cast.ToString(threshold.Operator),
-		Threshold:  cast.ToFloat64(threshold.Threshold),
-		Units:      cast.ToString(threshold.Units),
-		Mode:       cast.ToString(threshold.Mode),
+		Operator:   threshold.Operator,
+		Threshold:  threshold.Threshold,
+		Units:      threshold.Units,
+		Mode:       threshold.Mode,
 	}
 }
 
-func expandAlertConfigurationThreshold(threshold *IntegerThresholdView) *mongodbatlas.Threshold {
+func expandAlertConfigurationThreshold(threshold *IntegerThresholdView) *atlasSDK.GreaterThanRawThreshold {
 	if threshold == nil {
 		return nil
 	}
-	return &mongodbatlas.Threshold{
-		Operator:  cast.ToString(threshold.Operator),
-		Threshold: cast.ToFloat64(threshold.Threshold),
-		Units:     cast.ToString(threshold.Units),
+	return &atlasSDK.GreaterThanRawThreshold{
+		Operator:  threshold.Operator,
+		Threshold: util.Pointer(int(*threshold.Threshold)),
```

Comment:
we have a helper function in terraform called Int64PtrToIntPtr (would not apply to this case, but very similar) which does this operation safely.

As an additional point, since this method together with TimePtrToStringPtr are used in both terraform and CFN (related to the sdk) I believe we can take the opportunity to define them in the sdk directly. Let me know what you think, we can follow up with a ticket for this.

Reply:
Agreed, no concerns on my end. 
I will keep this PR open, in the meantime, create a ticket for the SDK to include these helper functions, I will assign that to myself and prioritize it so we are not delayed on waiting for these methods and can continue with rest of the migrations. 

@AgustinBettati @wtrocki Let me know if that works or any concerns here?

Reply:
Agreed, no concerns on my end, raised https://github.com/mongodb/atlas-sdk-go/pull/202 to add these utility methods. We can decide when to include these in our resources.

cc @wtrocki